src/main/java/com/redhat/rhjmc/containerjfr/net/SslConfiguration.java

```diff
@@ -0,0 +1,207 @@
+                return;
+            }
+        }
+
+        keystorePath = null;
+        keystorePass = null;
+        keyPath = null;
+        certPath = null;
+        logger.warn("No available SSL certificates. Fallback to plain HTTP.");
+    }
+
+    Path obtainKeyStorePathIfSpecified() throws IllegalArgumentException {
+        if (!env.hasEnv(KEYSTORE_PATH_ENV)) {
+            return null;
+        }
+
+        Path path = Path.of(env.getEnv(KEYSTORE_PATH_ENV)).normalize();
+        if (!fs.exists(keystorePath)) {
+            throw new IllegalArgumentException("KEYSTORE_PATH refers to a non-existent file");
+        }
+
+        return path;
+    }
+
+    Pair<Path, Path> obtainKeyCertPathPairIfSpecified() throws IllegalArgumentException {
+        if (!env.hasEnv(KEY_PATH_ENV) && !env.hasEnv(CERT_PATH_ENV)) {
+            return null;
+        }
+
+        if (env.hasEnv(KEY_PATH_ENV) ^ env.hasEnv(CERT_PATH_ENV)) {
+            throw new IllegalArgumentException("both KEY_PATH and CERT_PATH must be specified");
+        }
+
+        Path key = Path.of(env.getEnv(KEY_PATH_ENV)).normalize();
+        Path cert = Path.of(env.getEnv(CERT_PATH_ENV)).normalize();
+        if (!fs.exists(key)) {
+            throw new IllegalArgumentException("KEY_PATH refers to a non-existent file");
+        }
+
+        if (!fs.exists(cert)) {
+            throw new IllegalArgumentException("CERT_PATH refers to a non-existent file");
+        }
+
+        return new Pair<>(key, cert);
+    }
+
+    Path discoverKeyStorePathInDefaultLocations() {
+        Path home = Path.of(System.getProperty("user.home", "/"));
+
+        if (fs.exists(home.resolve("container-jfr-keystore.jks"))) {
+            return home.resolve("container-jfr-keystore.jks");
+        } else if (fs.exists(home.resolve("container-jfr-keystore.pfx"))) {
+            return home.resolve("container-jfr-keystore.pfx");
+        } else if (fs.exists(home.resolve("container-jfr-keystore.p12"))) {
+            return home.resolve("container-jfr-keystore.p12");
+        }
+
+        return null;
+    }
+
+    Pair<Path, Path> discoverKeyCertPathPairInDefaultLocations() {
+        Path home = Path.of(System.getProperty("user.home", "/"));
+        Pair<Path, Path> ret = new Pair<>(null, null);
+
+        // discover keyPath
+        if (fs.exists(home.resolve("container-jfr-key.pem"))) {
+            ret.left = home.resolve("container-jfr-key.pem");
+        } else {
+            logger.warn("unable to locate container-jfr-key.pem");
+        }
+
+        // discover certPath
+        if (fs.exists(home.resolve("container-jfr-cert.pem"))) {
+            ret.right = home.resolve("container-jfr-cert.pem");
+        } else {
+            logger.warn("unable to locate container-jfr-cert.pem");
+        }
+
+        if (keyPath == null || certPath == null) {
+            return null;
+        }
+
+        return ret;
+    }
+
+    HttpServerOptions setToHttpServerOptions(HttpServerOptions options) {
```

Comment:
`setTo` sounds kinda funny. `applyTo` maybe? Not too important.

If it mutates the HttpServerOptions that is passed in to it, does it need to also return back that same reference?

This is another place that looks like it could use some refactoring - there's a good chunk of nested branching logic in here.

Reply:
Yes, it mutates the `HttpServerOptions` passed it. `HttpServerOptions` implements fluent API so what's returned and what's passed in are actually the same reference.

Reply:
Refactored and adopted the strategy pattern.